packages/astro/src/core/build/generate.ts

```diff
@@ -518,10 +565,29 @@ async function generatePath(
 			}
 			throw err;
 		}
-		throwIfRedirectNotAllowed(response, opts.settings.config);
-		// If there's no body, do nothing
-		if (!response.body) return;
-		body = await response.text();
+
+		switch(true) {
+			case (response.status >= 300 && response.status < 400): {
```

Comment:
Why not an `if/else`?

Reply:
Not sure, can change it if that's the preference. My assumption was that this might expand to more status code handling but that might not be the case.